lib/ansible/modules/cloud/online/online_server_facts.py

```
@@ -0,0 +1,175 @@
+                    "reverse": "195-154-172-149.rev.poneytelecom.eu.",
+                    "switch_port_state": "up",
+                    "type": "public"
+                },
+                {
+                    "address": "10.90.53.212",
+                    "mac": "28:92:4a:33:5e:c7",
+                    "reverse": null,
+                    "switch_port_state": "up",
+                    "type": "private"
+                }
+            ],
+            "last_reboot": "2018-08-23T08:32:03.000Z",
+            "location": {
+                "block": "A",
+                "datacenter": "DC3",
+                "position": 19,
+                "rack": "A23",
+                "room": "4 4-4"
+            },
+            "network": {
+                "ip": [
+                    "195.154.172.149"
+                ],
+                "ipfo": [],
+                "private": [
+                    "10.90.53.212"
+                ]
+            },
+            "offer": "Pro-1-S-SATA",
+            "os": {
+                "name": "FreeBSD",
+                "version": "11.1-RELEASE"
+            },
+            "power": "ON",
+            "proactive_monitoring": false,
+            "raid_controllers": [
+                {
+                    "$ref": "/api/v1/server/hardware/raidController/9910"
+                }
+            ],
+            "support": "Basic service level"
+        }
+    ]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves.urllib.parse import urljoin
+from ansible.module_utils.online import (
+    Online, OnlineException, online_argument_spec
+)
+
+
+class OnlineServerFacts(Online):
+
+    def __init__(self, module):
+        super(OnlineServerFacts, self).__init__(module)
+        self.name = 'api/v1/server'
+
+    def _get_server_detail(self, server_path):
+        try:
+            return self.get(path=server_path).json
+        except OnlineException:
```

I think that the whole playbook should fail if the get fail, rather than silently forgetting some information.

---

Fixed it raises an OnlineException now